### ISSUE TITLE
happi and python 3.x

### ISSUE THREAD

**Issue by BenK3nobi** [SmileiPIC/Smilei#86]
Hello, 
I just started using smilei and get a lot of deprecated commands when using happi for plotting simulations results while using it via python 3
This is one of the errormessages: 

> python3.5/site-packages/matplotlib/axes/_base.py:3604: MatplotlibDeprecationWarning: 
> The `ymin` argument was deprecated in Matplotlib 3.0 and will be removed in 3.2. Use `bottom` instead.

I suppose the warning is a result of the newer python version. Are there any updates of happi planned to deal with this in the long term? 
Like Updating happi?
If no, how to participate?

**Comment by mccoys** [comment]
The warnings are due to a newer version of matplotlib, which is actually changing very fast. It is hard to keep up with all the changes while ensuring backward compatibility.

Happi is regularly updated, and will eventually take care of these warnings. I guess nobody developing smilei has switched to matplotlib 3 yet.

If you wish to participate, you may fork the smilei repository, then apply your changes, and submit a pull request to the base repo so that the admins may verify and include them.

**Comment by BenK3nobi** [comment]
Ah thanks a lot for the quick response. 
Well if I get some new stuff implemented I will definitely supply it. Since I just started with smilei, it might still take some time ;) 

One more quick question:
If I want to avoid these "errors" which matplotlib version is recommended? This might also be included in the manual at some point (or it is already there and I just did not find it).

**Comment by BenK3nobi** [comment]
Addendum: It appears that everything works fine with the matplotlib 2.0.0 version, in case anyone else might need this informaion.

**Comment by mccoys** [comment]
Yes, matplotlib 2 is the version I use. There might be some warnings left. However, warnings in matplotlib 3 should not make happi crash. They are only inconvenient.

There is no recommended version because there is no known bug with any of them. Warnings are not really a bug.

**Comment by BenK3nobi** [comment]
I actually could not get all of happis commands running with matplotlib 3- 
At least the animate command did not work. I assume it crashed. 
Now same code with matplotlib 2: No Problems.

**Comment by mccoys** [comment]
Ok, so there really must be an error message somewhere. I'll investigate

**Comment by mccoys** [comment]
@BenK3nobi could you provide an example which failed ? I managed to reproduce the warning, but no error.

**Comment by iltommi** [comment]
I've reproduced this on mac:
```
In [6]: s=happi.Open(".")                                                       
Loaded simulation '.'
Scanning for Scalar diagnostics
Scanning for Field diagnostics
Scanning for Probe diagnostics
/usr/local/lib/python3.7/site-packages/h5py/_hl/dataset.py:313: H5pyDeprecationWarning: dataset.value has been deprecated. Use dataset[()] instead.
  "Use dataset[()] instead.", H5pyDeprecationWarning)
Scanning for ParticleBinning diagnostics
Scanning for Screen diagnostics
Scanning for Tracked particle diagnostics
```
with `Python 3.7.2`  and
```
In [4]: h5py.__version__                                                        
Out[4]: '2.9.0'
````

**Comment by BenK3nobi** [comment]
Sorry it took me so long, I had some other issues to attend to. 
I am currently trying to reproduce the issue, which I solved by installing an earlier version of matplotlib.

**Comment by BenK3nobi** [comment]
So if I see it correctly, then these are the two files that caused the issue. 
One is the input file. It is basically one of the example files I started to modify for ion acceleration. 
The caculation went fine and the issue showed up when using happi in the evaluation script given here as well. 

Problem was solved through converting from matplotlib 3 to the one officially supported by smilei (2.x). 
[plasma.zip](https://github.com/SmileiPIC/Smilei/files/3019199/plasma.zip)

**Comment by mccoys** [comment]
Could you specify the error you obtain with matplotlib 3.0 ?

I think I have recently fixed an issue with plotting tracked particles, but this was not pushed to GitHub yet.

I tried your case and this is the error I obtain:
```
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
/home/fperez/Repos/EvalPlasmaExp.py in <module>()
     71 PartEon = S.TrackParticles(species="eon", select="", axes=["x", "px"], timesteps=None, sort=True, length=None, units=[""])
     72 
---> 73 PlotIon = PartIon - S.TrackParticles(species="ion", select="", axes=["x", "Ex"], timesteps=None, sort=True, length=None, units=[""])
     74 
     75 

TypeError: unsupported operand type(s) for -: 'TrackParticles' and 'TrackParticles'
```

This is due to a wrong subtraction (diagnostics cannot be subtracted).
